debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/ValueTypeHelper.java

```diff
@@ -0,0 +1,392 @@
+		}
+		
+		@Override
+		public J9ClassPointer[] findNestedClassHeirarchy(J9ClassPointer containerClazz, String[] nestingHierarchy) throws RuntimeException {
+			logger.log(Level.SEVERE, "Value types were not enabled in the JVM that produced this core file.");
+			throw new RuntimeException("Invalid operation");
+		}
+		
+		@Override
+		public J9ClassPointer findJ9ClassInFlattenedClassCacheWithSigName(J9ClassPointer clazz, String fieldSig) throws RuntimeException {
+			logger.log(Level.SEVERE, "Value types were not enabled in the JVM that produced this core file.");
+			throw new RuntimeException("Invalid operation");
+		}
+		
+		@Override
+		public boolean isFieldInClassFlattened(J9ClassPointer clazz, String fieldName) throws RuntimeException {
+			return false;
+		}
+		
+		@Override
+		public boolean areValueTypesSupported() {
+			return false;
+		}
+		
+		@Override
+		public boolean isRomClassAValueType(J9ROMClassPointer romClass) throws RuntimeException {
+			return false;
+		}
+		
+		@Override
+		public boolean isJ9ClassAValueType(J9ClassPointer clazz) throws RuntimeException {
+			return false;
+		}
+		
+		@Override
+		public boolean isFlattenableFieldSignature(String signature) {
+			return false;
+		}
+	}
+	
+	
+	private ValueTypeHelper() {
+		try {
+			Lookup lookup = MethodHandles.lookup();
+			flattenedClassCachePointer = (Class<?>) Class.forName("com.ibm.j9ddr.vm29.pointer.generated.J9FlattenedClassCachePointer");
+			getFlattenedClassCachePointer = lookup.findVirtual(J9ClassPointer.class, "flattenedClassCache", MethodType.methodType(flattenedClassCachePointer));
+			flattenedClassCache_offset = lookup.findVirtual(flattenedClassCachePointer, "offset", MethodType.methodType(UDATA.class));
+			flattenedClassCache_NaS = lookup.findVirtual(flattenedClassCachePointer, "nameAndSignature", MethodType.methodType(J9ROMNameAndSignaturePointer.class));
+			flattenedClassCache_clazz = lookup.findVirtual(flattenedClassCachePointer, "clazz", MethodType.methodType(J9ClassPointer.class));
+			flattenedClassCache_cast = lookup.findStatic(flattenedClassCachePointer, "cast", MethodType.methodType(flattenedClassCachePointer, AbstractPointer.class));
+		} catch (Throwable t) {
+			/* should not happen */
+			throwUncheckedExceptions(t);
+			logger.log(Level.SEVERE, "ValueTypeHelper: failed to get ValueTypeHelper method handles", t);
+		}
+	}
+	
+	public static ValueTypeHelper getValueTypeHelper() {
+		if (helper == null) {
+			if (checkIfValueTypesAreSupported()) {
+				helper = new ValueTypeSupportEnabledHelper();
+			} else {
+				helper = new ValueTypeSupportDisabledHelper();
+			}
+		}
+		return helper;
+	}
+	
+	private static void throwUncheckedExceptions(Throwable t) {
+		if (t instanceof Error) {
+			throw (Error)t;
+		} else if (t instanceof RuntimeException) {
+			throw (RuntimeException)t;
+		}
+	}
+	
+	private static CorruptDataException handleThrowable(Throwable t) {
+		throwUncheckedExceptions(t);
+		return new CorruptDataException(t);
```

Comment:
This should be `throw` instead of `return`; the return type can stay as is and the callers can be left as they are.